### PR TITLE
Upgrade to gradle 5.6.4

### PR DESCRIPTION
**Pull request by Will-Lo**
Dear Gobblin maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- [x] My PR addresses the following [Gobblin JIRA](https://issues.apache.org/jira/browse/GOBBLIN/) issues and references them in the PR title. For example, "[GOBBLIN-XXX] My Gobblin PR"
    - https://issues.apache.org/jira/browse/GOBBLIN-1293

### Description
- [x] Here are some details about my PR, including screenshots (if applicable):

Upgrades from Grade 4.9 to 5.6.4. The previous attempt to upgrade failed as it upgraded to version 5.6.0 and it introduced a regression where it would fail (instead of warn) when overwriting one package with another during publishing.

### Tests
- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

Tested with multiple clean builds, and maven publishing

### Commits
- [ ] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,24 +13,10 @@ Apache Gobblin handles the common routine tasks required for all data ingestion
...
-wget --no-check-certificate -P gradle/wrapper https://github.com/apache/incubator-gobblin/raw/0.12.0/gradle/wrapper/gradle-wrapper.jar
-(or)
-curl --insecure -L https://github.com/apache/incubator-gobblin/raw/0.12.0/gradle/wrapper/gradle-wrapper.jar > gradle/wrapper/gradle-wrapper.jar
-
-Alternatively, you can download it manually from: 
-https://github.com/apache/incubator-gobblin/blob/0.12.0/gradle/wrapper/gradle-wrapper.jar
-
-Make sure that you download it to gradle/wrapper directory. 
-
+* Maven version 3.5.
```
Comment (autumnust) [inline README.md:19]: 3.5. ?

**Review by autumnust — CHANGES_REQUESTED** [review]
LGTM overall. 
How does the publish to bintray tested?

File: README.md (hunk 1 above)
Comment (arjun4084346) [inline README.md:19]: Is this deletion intentional?

File: gobblin-api/build.gradle [hunk 2]
```diff
@@ -29,7 +29,11 @@ dependencies {
     compile externalDependency.reflections
     compile externalDependency.typesafeConfig
     compile externalDependency.reactivex
-
+    // Modified to work with Gradle 5
+    compileOnly externalDependency.lombok
```
Comment (arjun4084346) [inline gobblin-api/build.gradle:33]: Can the common code be moved to outer build.gradle?

**Comment by Will-Lo** [comment]
@autumnust I was testing with publishToMavenLocal so far

**Comment by Will-Lo** [comment]
Also tested with dryruns and with a test bintray account.

**Pull request closed without merging**